src/jsifier.js

```diff
@@ -108,46 +108,63 @@ function runJSify() {
-    // to double (this limits the range to int53).
-    // And convert the return value if the function returns a pointer.
-    return modifyJSFunction(snippet, (args, body) => {
+  function handleI64Signatures(symbol, snippet, sig, i53abi) {
+    // Handler i64 paramaters and return values.
+    //
+    // When WASM_BIGINT is enabled these arrive as BigInt values which we
+    // convert to int53 JS numbers.  If necessary, we also convert the return
+    // value back into a BigInt.
+    //
+    // When WASM_BIGINT is not enabled we receive i64 values as a pair of i32
+    // numbers which is coverted to single int53 number.  In necessary, we also
+    // split the return value into a pair of i32 numbers.
+    return modifyJSFunction(snippet, (args, body, async_) => {
       let argLines = args.split('\n');
       argLines = argLines.map((line) => line.split('//')[0]);
       const argNames = argLines.join(' ').split(',').map((name) => name.trim());
-      let newArgs = [];
+      const newArgs = [];
+      let innerArgs = [];
       let argConvertions = '';
       for (let i = 1; i < sig.length; i++) {
         const name = argNames[i - 1];
         if (!name) {
           error(`convertPointerParams: missing name for argument ${i} in ${symbol}`);
           return snippet;
         }
-        if (sig[i] == 'p') {
-          argConvertions += `  ${name} = Number(${name});\n`;
-          newArgs.push(`Number(${name})`);
+        if (WASM_BIGINT) {
+          if (sig[i] == 'p' || (sig[i] == 'j' && i53abi)) {
+            argConvertions += `  ${receiveI64ParamAsI53(name, undefined, false)}\n`;
+          }
         } else {
-          newArgs.push(name);
+          if (sig[i] == 'j' && i53abi) {
+            argConvertions += `  ${receiveI64ParamAsI53(name, undefined, false)}\n`;
+            newArgs.push(defineI64Param(name));
+          } else {
+            newArgs.push(name);
+          }
         }
       }
 
-      if (sig[0] == 'p') {
-        // For functions that return a pointer we need to convert
-        // the return value too, which means we need to wrap the
-        // body in an inner function.
-        newArgs = newArgs.join(',');
+      var origArgs = args;
+      if (!WASM_BIGINT) {
+        args = newArgs.join(',');
+      }
+
+      if ((sig[0] == 'j' && i53abi) || (sig[0] == 'p' && WASM_BIGINT)) {
+        // For functions that where we need to mutate the return value, we
+        // also need to wrap the body in an inner function.
         return `\
-function(${args}) {
-  var ret = ((${args}) => { ${body} })(${newArgs});
-  return BigInt(ret);
+${async_}function(${args}) {
+${argConvertions}
+  var ret = (() => { ${body} })();
+  return ${makeReturn64('ret')};
 }`;
       }
 
       // Otherwise no inner function is needed and we covert the arguments
       // before executing the function body.
       return `\
-function(${args}) {
+${async_}function(${args}) {
```

Comment:
As the `async` change related?

Reply:
Yes, all calls to `modifyJSFunction` should really handle the final `async_` argument like this.    In this case we were previously getting away without handling it because were not running this modifier on any async functions.  With this change we are.